### ISSUE TITLE
Compiler should check custom component tags are closed

### ISSUE THREAD

**Issue by robwormald** [angular/angular#5247]
Seen a few usages of this already:
```
<my-custom-element />
```
HTML5 spec says this is opening tag (previous versions handled it differently) and thus this fails silently and swallows up any following DOM. Compiler should probably warn / throw here if possible.

**Comment by pkozlowski-opensource** [comment]
We can't do this with the current compiler as it works off the DOM so by the time compiler kicks in the HTML from your example is transformed to `<my-custom-element></my-custom-element>` by a browser - and this is done before we can do anything on the ng2 side.

A new text-based compiler should be able to handle it though (either by accepting self-closing tags or by warning people about them). 

Closing as duplicate of #4417
